src/vm/method.cpp

```diff
@@ -3684,16 +3682,28 @@ void MethodDesc::SaveChunk::SaveOneChunk(COUNT_T start, COUNT_T count, ULONG siz
         UNREACHABLE();
     }
 
-    ULONG size = sizeOfMethodDescs + sizeof(MethodDescChunk);
-    ZapStoredStructure * pNode = m_pImage->StoreStructure(NULL, size, kind);
-
-    BYTE * pData = (BYTE *)m_pImage->GetImagePointer(pNode);
+    ULONG size = sizeOfMethodDescs + sizeof(MethodDescChunk) + sizeof(MethodDescChunk::TemporaryEntryPointsSlot);
+    ZapStoredStructure * pStructureNode = m_pImage->StoreStructure(NULL, size, kind);
 
-    MethodDescChunk * pNewChunk = (MethodDescChunk *)pData;
+    BYTE * pData = (BYTE *)m_pImage->GetImagePointer(pStructureNode);
 
+    MethodDescChunk::TemporaryEntryPointsSlot * pTemporaryEntryPointsSlot =
+        (MethodDescChunk::TemporaryEntryPointsSlot *)pData;
     // Bind the image space so we can use the regular fixup helpers
+    m_pImage->BindPointer(pTemporaryEntryPointsSlot, pStructureNode, 0);
+
+    MethodDescChunk * pNewChunk = (MethodDescChunk *)(pData + sizeof(MethodDescChunk::TemporaryEntryPointsSlot));
+    // Don't bind the chunk pointer to the structure node since that would mean
+    // it doesn't start at offset 0. Instead create an inner ptr node and bind to that one.
+    // This helps with not spreading the knowledge of the storage of the temporary entry points slot
+    // throughout the codebase (since several pieces need to be able to map into the chunks
+    // during fixups).
+    ZapNode * pNode = m_pImage->GetInnerPtr(pStructureNode, sizeof(MethodDescChunk::TemporaryEntryPointsSlot));
```

Comment:
I actually started with that.. 
Unfortunately there are several places where we ignore the offset part of the pointer->node lookup. Specifically in FixupPrecode this would introduce a bit more mess into the code. So I added the innerptr node - which worked well, but then broken in other places - but overall it didn't feel that bad.
It's one or the other - more complexity in the infra, for the benefit of keeping "users" cleaner. And also unaware of the actual layout of the method desc chunk and more importantly it's ZapNode storage details.
Or - less complex infra, but more knowledge of zap node storage layout outside of the MD chunk.
I personally prefer the approach with the inner ptr, but if you think the other way would be better I can definitely switch it.

Reply:
I think my preference would be to do it the other way. I think it is better to keep the magic in the places that need them, not push them into the infra piece.